examples/put_to_self.c

```diff
@@ -64,73 +65,94 @@ int main(void)
-			NULL, &retNIInit);
+			NULL, &nih);
 	if (ret != PTL_OK) {
 		fprintf(stderr, "PtlNIInit failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
-		return 1;
+		res = 1;
+		goto fini;
 	}
 
-	ret = PtlEQAlloc(retNIInit, 10, &retEQAlloc);
+	ret = PtlEQAlloc(nih, 10, &eqh);
 	if (ret != PTL_OK) {
 		fprintf(stderr, "PtlEQAlloc failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
-		return 1;
+		res = 1;
+		goto ni_fini;
 	}
 
-	ret = PtlPTAlloc(retNIInit, PTL_PT_FLOWCTRL, retEQAlloc, PTL_PT_ANY, &retPTAlloc);
+	ret = PtlPTAlloc(nih, PTL_PT_FLOWCTRL, eqh, PTL_PT_ANY, &pti);
 	if (ret != PTL_OK) {
 		fprintf(stderr, "PtlPTAlloc failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
-		return 1;
+		res = 1;
+		goto eq_free;
 	}
 
-	ret = PtlGetId(retNIInit, &id);
+	ret = PtlGetId(nih, &id);
 	if (ret != PTL_OK) {
 		fprintf(stderr, "PtlGetId failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
 		return 1;
 	}
 
-	/*prepare our events buffer to receive a message represented by le*/
-	ret = PtlLEAppend(retNIInit, retPTAlloc, &le, PTL_PRIORITY_LIST, NULL, &retAppend);
+	/* expose to the network a portion of the process address space */
+	ret = PtlLEAppend(nih, pti, &le, PTL_PRIORITY_LIST, NULL, &leh);
 	if (ret != PTL_OK) {
 		fprintf(stderr, "PtlLEAppend failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
-		return 1;
+		res = 1;
+		goto pt_free;
 	}
 
-	/*wait for the event link*/
-	ret = PtlEQWait(retEQAlloc, &retWait);
+	ret = PtlEQWait(eqh, &ev);
 	if (ret != PTL_OK) {
 		fprintf(stderr, "PtlEQWait failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
-		return 1;
-	} else {
-		PtlEvToStr(0, &retWait, msg);
-		printf("Event : %s", msg);
+		res = 1;
+		goto le_unlink;
 	}
+	PtlEvToStr(0, &ev, msg);
+	printf("Event : %s \n", msg);
 
-	md.eq_handle = retEQAlloc;
-	/*allocate a buffer for our message represented by md*/
-	ret = PtlMDBind(retNIInit, &md, &retBind);
+	md.eq_handle = eqh;
+	/* create a memory descriptor to be used to send message */
+	ret = PtlMDBind(nih, &md, &mdh);
 	if (ret != PTL_OK) {
 		fprintf(stderr, "PtlMDBind failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
-		return 1;
+		res = 1;
+		goto le_unlink;
 	}
 
-	/*send the message to ourself*/
-	ret = PtlPut(retBind, 0, sizeof(data), PTL_ACK_REQ, id, retPTAlloc, 0, 0, NULL,
-		     header_data);
+	/* send the message to ourself */
+	ret = PtlPut(mdh, 0, sizeof(data), PTL_ACK_REQ, id, pti, 0, 0, NULL, header_data);
 	if (ret != PTL_OK) {
 		fprintf(stderr, "PtlPut failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
-		return 1;
+		res = 1;
+		goto md_release;
 	}
 
 	/* Wait for the send, ack & put events */
 	for (int i = 0; i < 3; i++) {
-		ret = PtlEQWait(retEQAlloc, &retWait);
+		ret = PtlEQWait(eqh, &ev);
 		if (ret != PTL_OK) {
 			fprintf(stderr, "PtlEQWait failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
-		} else {
-			PtlEvToStr(0, &retWait, msg);
-			printf("Event : %s", msg);
+			res = 1;
+			goto md_release;
 		}
+		res = 0;

```

Comment:
having "res" variable initialized to 0 at declaration times will avoid this line of code being potentially executed 3 times.